File: bookOfProblems/leetcode.com/javascript/217. Contains Duplicate.js
let nums = [1, 2, 2]

const containsDuplicate = nums => {
    let seen = new Map();
    for (let num = 0; num < nums.length; num++) {
        if (seen.has(nums[num])) return true;
        seen.set(nums[num]);
    }
    return false;
};

console.log(containsDuplicate(nums));

////////////////////////////////////////////////////////////////////////////////

var containsDuplicate = function(nums) {
    return new Set(nums).size !== nums.length
};

File: bookOfProblems/leetcode.com/golang/hashtable/217. Contains Duplicate.go
func containsDuplicate(nums []int) bool {
	numSet := make(map[int]struct{})

	for _, num := range nums {
		if _, found := numSet[num]; found {
			return true
		}
		numSet[num] = struct{}{}
	}

	return false
}

////////////////////////////////////////////////////////////////////////////////

func containsDuplicate(nums []int) bool {
	q := make(map[int]int)

	for i := 0; i < len(nums); i++ {
		q[nums[i]]++
		if (q[nums[i]]) > 1 {
			return true
		}
	}

	return false
}

////////////////////////////////////////////////////////////////////////////////

func containsDuplicate(nums []int) bool {
    
    m := make(map[int]bool, len(nums))
	for _, n := range nums {
		if _, ok := m[n]; ok {
			return true
		} else {
			m[n] = true
		}
	}
	return false
}